pkg/promutil/prometheus.go

```diff
@@ -155,10 +154,54 @@ func PromStringTag(text string, labelsSnakeCase bool) (bool, string) {
 	return model.LabelName(s).IsValid(), s
 }
 
+// sanitize replaces some invalid chars with an underscore
 func sanitize(text string) string {
-	return replacer.Replace(text)
+	if strings.ContainsAny(text, "“%") {
+		// fallback to the replacer for complex cases:
+		// - '“' is non-ascii rune
+		// - '%' is replaced with a whole string
+		return replacer.Replace(text)
+	}
+
+	b := []byte(text)
+	for i := 0; i < len(b); i++ {
+		switch b[i] {
+		case ' ', ',', '\t', '/', '\\', '.', '-', ':', '=', '@', '<', '>':
+			b[i] = '_'
+		}
+	}
+	return *(*string)(unsafe.Pointer(&b))
```

Comment:
this is more efficient than doing `string(b)`?

Reply:
Good point, I don't think unsafe is needed anymore, I was playing with it in a previous iteration. Shouldn't change much in terms of performance, but `string(b)` is easier to read.